overlord/snapstate/autorefresh_test.go

```diff
@@ -49,6 +49,10 @@ type autoRefreshStore struct {
 }
 
 func (r *autoRefreshStore) SnapAction(ctx context.Context, currentSnaps []*store.CurrentSnap, actions []*store.SnapAction, user *auth.UserState, opts *store.RefreshOptions) ([]*snap.Info, error) {
+	if !opts.IsAutoRefresh {
+		panic("AutoRefresh snap action did not set IsAutoRefresh flag")
```

Comment:
Slightly surprised to see panic here, I'd expect an internal error (apparently this is the approach taken in the existing checks below).

Reply:
surprising *and* unhelpful, so I'll probably be changing it in a separate PR.

As it stands, if any of these checks fail, you get a very unhelpful panic about the state lock, not about the actual error.